components/bases-locales/editor/edit-bal/search-communes.js

```diff
@@ -54,9 +62,12 @@ class SearchCommune extends React.Component {
   }
 
   async handleSearch(input) {
+    const codeDep = input.split(' ').find(isCodeDepNaive)
     const reqId = uniqueId('req_')
     this.waitingFor = reqId
-    const url = `https://geo.api.gouv.fr/communes?nom=${input}&fields=departement,contour&limit=5`
+    const codeDepFilter = codeDep ? `&codeDepartement=${codeDep}` : ''
+    const q = codeDep ? input.split(' ').filter(t => !isCodeDepNaive(t)).join(' ') : input
+    const url = `https://geo.api.gouv.fr/communes?nom=${q}${codeDepFilter}&fields=departement,contour&limit=8`
```

Comment:
I’d use [URLSearchParams](https://nodejs.org/api/url.html#url_class_urlsearchparams) to build the url, `q` could be unsafe.

Reply:
The user may alter town selector results quality…
Unsafe but not very harmful.

Reply:
At least use encodeURIComponent

Reply:
Yes but give me an harmful example ;)

Reply:
par&size
p&size=20000


I don’t know all the query params.

Reply:
g&format=geojson – try this one, it’s pretty good